packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js

```diff
@@ -85,7 +85,7 @@ function linkHeaders(files, pathPrefix, assetPrefix) {
     files[resourceType].forEach(file => {
       linkHeaders.push(
         linkTemplate(
-          `${assetPrefix && assetPrefix + `/`}${pathPrefix}/${file}`,
+          `${assetPrefix ? assetPrefix + `/` : ``}${pathPrefix}/${file}`,
```

Comment:
This was producing `undefined/foo/bar` kind of results when `assetPrefix` was falsy